tests/Main.hs

```diff
@@ -662,13 +664,21 @@ runClashTest = defaultMain $ clashTestRoot
                                                         , "testBenchUS"
                                                         ]}
           in runTest "DDRin" _opts
+
+          -- XXX: `ddrOut` contains two concurrent always-blocks in the generated
+          --      Verilog. Execution order is not defined in the Verilog spec, so
+          --      simulators are free to pick an order. ModelSim and Verilator
+          --      disagree on the order, so when a test works in one, it doesn't
+          --      in the other. As a quick "fix" we disable Verilator, though we
```

Comment:
As discussed on slack, the problem isn't with the two always-blocks.
But it's between one of the always blocks and the continuous assignment of the result.
(To add to the confusion in SystemVerilog the assignment of the result is done in a third always-block instead. So there the problem is between two (out of three) always-blocks.)

You should probably make the description here just slightly more general.

Reply:
Yes, I'll generalize the description. Luckily Nix failed :-)